distributedcombigrid/src/sgpp/distributedcombigrid/task/Task.hpp

```diff
@@ -110,6 +110,38 @@ class Task {
   LoadModel* loadModel_;
 
   bool isFinished_;
+
+  public:
+  /** the full grid vector before combi, this contains the elements of the full grid before combi*/
+  std::vector<CombiDataType> *fullgridVectorBeforeCombi;
+
+  /** flags if the combisequence is happening for the first time.*/
+  static bool isFirstCombiSequence;
+
+  /** Buffer for Async Reduce*/
+  static std::vector<CombiDataType> *bufAsync;
+
+  /** request for Async Reduce*/
+  static MPI_Request *requestAsync;
+
+  /** SubspaceSized for Async Reduce*/
+  static std::vector<int> *subspaceSizes;
```

Comment:
I feel like this should not be a static member of `Task` because it already exists in `DistributedSparseGridUniform`, and I think that this is where it should be.

Reply:
As far as I know this is in task because the Sparse Grid gets destroyed frequently but this information should be constant over multiple combinations.

Reply:
But this is just an optimization so that the sparse grid sizes do not need to be communicated again. For the request array I would need to investigate if it is necessary to be in task.

Reply:
But one could also put the subspaceSizes as static member of the SparseGrid class I think...

Reply:
Yes, you are right, we can leave it like this here. on `third-level`, `subspacesDataSizes_` is introduced which makes this redundant. I had not noticed that it wasn't on master.